Frends.PowerShell.RunCommand/Frends.PowerShell.RunCommand/PowerShell.cs

```diff
@@ -34,9 +36,81 @@ public static PowerShellResult RunCommand(RunCommandInput input, [Browsable(fals
     {
         return DoAndHandleSession(options?.Session, (session) =>
         {
-            return ExecuteCommand(input.Command, input.Parameters, input.LogInformationStream, session.PowerShell);
+            if (input.ExecuteNativeShell)
+            {
+                var tempScript = Path.Combine(Path.GetTempPath(), $"{Path.GetRandomFileName()}.ps1");
+                try
+                {
+                    string parameterString = string.Empty;
+                    if (input.Parameters != null)
+                    {
+                        foreach (var param in input.Parameters)
+                        {
+                            parameterString += $"-{param.Name} \"{param.Value}\" ";
+                        }
+                    }
+                    File.WriteAllText(tempScript, $"{input.Command} {parameterString}", Encoding.UTF8);
+                    return ExecuteProcess(tempScript, input.Parameters);
+                }
+                finally
+                {
+                    File.Delete(tempScript);
+                }
+            }
+            else
+                return ExecuteCommand(input.Command, input.Parameters, input.LogInformationStream, session.PowerShell);
         });
     }
+    private static PowerShellResult ExecuteProcess(string scriptPath, PowerShellParameter[] parameters)
+    {
+        List<dynamic> results = new();
+        List<string> errors = new();
+
+
+
+        using Process process = new();
+        process.StartInfo = new()
+        {
+            FileName = "powershell.exe",
+            Arguments = $"-NoProfile -ExecutionPolicy Bypass -File \"{scriptPath}\" ",
+            UseShellExecute = false,
+            CreateNoWindow = true,
+            RedirectStandardError = true,
+            RedirectStandardOutput = true,
+        };
+
+        process.OutputDataReceived += (sender, args) =>
+        {
+            if (!string.IsNullOrWhiteSpace(args.Data))
+                results.Add(args.Data);
+        };
+
+        process.ErrorDataReceived += (sender, args) =>
```

Comment:
I think that we need to unwire event handlers for .NET to be able to clean up the `process` object. So those should be in the finally block.